PHAL/UART/phal_uart.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * phal_uart.h
+ *
+ *  Created on: Jan. 20, 2024
+ *      Author: manoj
+ *
+ *
+ */
+
+#ifndef PHALD_UART_PHAL_UART_H_
+#define PHALD_UART_PHAL_UART_H_
+
+#include "sci.h"
+#include "reg_sci.h"
+#include "sys_vim.h"
+#include "sys_core.h"
+#include <assert.h>
+
+typedef void (*uartRxCallback_t)(char);
+typedef void (*uartTxCallback_t)(void);
+
+typedef enum {
+    PHAL_UART_SCI_PORT = (uint32)0,
+    PHAL_UART_SCILIN_PORT = (uint32)1
+} uart_port_id;
+
+typedef enum  {
+    PHAL_UART_WAIT = (uint32)0,
+    PHAL_UART_FORCE = (uint32)1
+} uart_mode;
+
+typedef enum {
+    PHAL_UART_INTR_HIGH =  (uint32)0,
+    PHAL_UART_INTR_LOW =  (uint32)1,
+    PHAL_UART_INTR_DISABLE = (uint32)2
+} uart_intr_level;
+
+typedef struct {
+    uint32 priority_high;
+    uint32 priority_low;
+    uart_intr_level rx_level;
+    uart_intr_level tx_level;
+} uart_intr_setup;
```

Comment:
Can we get descriptions for these enums? Also with a lot of user defined types, it may be good to distinguish between enums and structs. I usually use this convention for the VCU codebase. Since this is a different repo I'm fine using a slightly different style for the drivers

```
eCarEvents
vs 
state_machine_t
```

Reply:
sure using camelCase for enums and snake_case_t for structs, fixed it here https://github.com/sfuphantom/hercules-phantom-lib/pull/13/commits/c40d17920878f843855f686fc55f11eeddee8a78

Reply:
oh sorry, I didn't clarify on the style. Enums start with an 'e' that's why eCarEvents and structs end with _t. Idk if that's the best convention but as long as there is one for this repo I don't mind what it is. Happy to resolve this thread as there is a convention now

Reply:
Changed the `async_*` functions name to `listen_for_*`.